apps/epic-react/src/data/purchase-import.ts

```diff
@@ -0,0 +1,297 @@
+      const chunks = chunk(data, 1000)
+
+      if (shortRun) {
+        return chunks.slice(0, 1)
+      } else {
+        return chunks
+      }
+    }
+
+    // These UUIDs have no significance and aren't related to existing data. They
+    // were generated to be used as a consistent ID value for the User and
+    // Customer records.
+    const nullUserId = 'EFAEA7AA-2D37-481B-883E-E2D0DAF5ACD4' as string
+    const nullCustomerId = 'kcd_50A6E8C7-0ECA-43C1-848B-91080EAD7B4D' as string
+
+    // Create a null customer if it doesn't already exist
+    await prisma.user.upsert({
+      where: {id: nullUserId},
+      update: {},
+      create: {
+        name: 'Null Customer',
+        email: 'nullcustomer@egghead.io',
+        emailVerified: null,
+      },
+    })
+    // Note: this references the `merchantAccountId` for the one _shared_ Stripe
+    // merchant account, so it should be the same across KCD products.
+    await prisma.merchantCustomer.upsert({
+      where: {
+        id: nullCustomerId,
+      },
+      update: {},
+      create: {
+        userId: nullUserId,
+        merchantAccountId: 'kcd_ff532118-69fe-4263-85a5-50b7b03a4b1e',
+        identifier: 'cus_kcd_null_customer',
+        createdAt: now,
+        status: 0,
+      },
+    })
+
+    for (const userChunk of chunkData(users)) {
+      await prisma.user.createMany({data: userChunk, skipDuplicates: true})
+    }
+
+    for (const couponChunk of chunkData(coupons)) {
+      await prisma.coupon.createMany({data: couponChunk, skipDuplicates: true})
+    }
+
+    for (const merchantCustomerChunk of chunkData(merchantCustomers)) {
+      await prisma.merchantCustomer.createMany({
+        data: merchantCustomerChunk,
+        skipDuplicates: true,
+      })
+    }
+
+    for (const merchantChargeChunk of chunkData(merchantCharges)) {
+      const fillInNullCustomer = (
+        merchantCharge: MerchantCharge,
+      ): CleanMerchantCharge => {
+        const validCustomerId =
+          merchantCharge.merchantCustomerId || nullCustomerId
+
+        return {
+          ...merchantCharge,
+          merchantCustomerId: validCustomerId,
+        }
+      }
+
+      const cleanedMerchantChargeChunk =
+        merchantChargeChunk.map(fillInNullCustomer)
+
+      await prisma.merchantCharge.createMany({
+        data: cleanedMerchantChargeChunk,
+        skipDuplicates: true,
+      })
+    }
+
+    for (const purchaseChunk of chunkData(purchases)) {
+      await prisma.purchase.createMany({
+        data: purchaseChunk,
+        skipDuplicates: true,
+      })
+    }
+  }
+
+  // ---------------------------------------
+  // Stream large file and parse to JSON
+  const pipeline = chain([
+    fs.createReadStream(dataFilePath),
+    parser(),
+    streamValues(),
+  ])
+
+  // Aggregate all the data and then process it (push to database)
+  pipeline.on('data', collectData)
+  pipeline.on('end', processData)
+  // ---------------------------------------
```

Comment:
pretty much same story in this file regarding the data be streamed in as JSON, see comments on the Lesson Progress import script for more details.